omniledger/service/struct.go

```diff
@@ -57,6 +57,28 @@ func (c *collectionDB) loadAll() {
 	})
 }
 
+// tryHash returns the merkle root of the collection as if the key value pairs
+// in the transactions had been added, without actually adding it.
+func (c *collectionDB) tryHash(ts []Transaction) (mr []byte, rerr error) {
+	for _, t := range ts {
+		err := c.coll.Add(t.Key, t.Value, t.Kind)
+		if err != nil {
+			rerr = err
+			return
+		}
+		// remove the pair after we got the merkle root.
+		defer func(k []byte) {
```

Comment:
I'm mitigated on this defer - on one hand it's a nice solution in case we get an error in the middle of the `for`, on the other hand it looks non-obvious to me and I would prefer having:
```
for range ts{
  Add
}
mr = c.coll.GetRoot
for range ts{
  Remove
}
```

@kc1212 - what do you think?